.circleci/config.yml

```diff
@@ -633,11 +747,536 @@ jobs:
+      - setup_remote_docker
+      - *restore_cache
+      - *prepare_environment
+      - save_cache:
+          paths:
+            - venv
+            - node_modules
+          key: venv-{{ checksum "dev-requirements-py2.txt" }}-{{ checksum "dev-requirements-py3.txt" }}-{{ checksum ".circleci/build-requirements.txt" }}-{{ checksum "package-lock.json" }}
+      - *add_ssh_keys
+      - *persist_to_workspace
+
+  Run Unit Testing And Lint:
+    <<: *container_config
+    resource_class: medium+
+    <<: *environment
+    steps:
+      - *attach_workspace
+      - setup_remote_docker
+      - *restore_cache
+      - *add_ssh_keys
+      - *prepare_environment
+      - run:
+          name: Infrastructure testing
+          when: always
+          command: |
+            python3 -m pytest ./Tests/scripts/infrastructure_tests/ -v
+            python3 -m pytest ./Tests/Marketplace/Tests/marketplace_services_test.py -v
+            python3 -m pytest ./Tests/Marketplace/Tests/upload_packs_test.py -v
+            python3 -m pytest ./Tests/Marketplace/Tests/search_and_install_packs_test.py -v
+      - run:
+          name: Run Unit Testing and Lint
+          when: always
+          no_output_timeout: 5h
+          command: |
+            echo "demisto-sdk version: $(demisto-sdk --version)"
+            echo "mypy version: $(mypy --version)"
+            echo "flake8 py2 version: $(python2 -m flake8 --version)"
+            echo "flake8 py3 version: $(python3 -m flake8 --version)"
+            echo "bandit py2 version: $(python2 -m bandit --version 2>&1)"
+            echo "bandit py3 version: $(python3 -m bandit --version 2>&1)"
+            echo "vulture py2 version: $(python2 -m vulture --version) 2>&1"
+            echo "vulture py3 version: $(python3 -m vulture --version) 2>&1"
+            SHOULD_LINT_ALL=$(./Tests/scripts/should_lint_all.sh)
+            mkdir ./unit-tests
+            if [ -n "$SHOULD_LINT_ALL" ]; then
+              echo -e  "----------\nLinting all because:\n${SHOULD_LINT_ALL}\n----------"
+              demisto-sdk lint -p 8 -a -q --test-xml ./unit-tests --log-path ./artifacts --failure-report ./artifacts
+            else
+              demisto-sdk lint -p 8 -g -v --test-xml ./unit-tests --log-path ./artifacts --failure-report ./artifacts
+            fi
+      - store_test_results:
+          path: ./unit-tests
+      - run:
+          name: Slack Notifier
+          shell: /bin/bash
+          command: |
+            ./Tests/scripts/slack_notifier.sh 'unittests' ./env_results.json
+          when: always
+      - *store_artifacts
+  Validations:
+    <<: *container_config
+    resource_class: medium+
+    <<: *environment
+    steps:
+      - *attach_workspace
+      - setup_remote_docker
+      - *restore_cache
+      - *add_ssh_keys
+      - *prepare_environment
+      - run:
+          name: Secrets
+          when: always
+          command: |
+            demisto-sdk secrets --post-commit --ignore-entropy
```

Comment:
do not run for forks

Reply:
Looking at it now- it seems like it should run for forks too
```
run:
          name: Secrets
          when: always
          command: |
            if [ -n "${INSTANCE_TESTS}" ] || [ $CIRCLE_NODE_INDEX -ne 0 ] ;
              then
                echo "Skipping - not running in INSTANCE_TESTS build or unit-tests container"
                exit 0
            fi

            demisto-sdk secrets --post-commit --ignore-entropy
```